### ISSUE TITLE
Compile 1.3.1 for Windows Pidgin - MinGW issues

### ISSUE THREAD

**Issue by JimmyS83** [majn/telegram-purple#383]
Hello,

I am trying to compile 1.3.1 for windows. But I cant use mingw makefile. My procedure is:

```
git clone --recursive https://github.com/majn/telegram-purple;cd telegram-purple
git submodule init
git submodule update --recursive
git reset --hard v1.3.1
./configure
make
```
Here I have non-problematic compilation for linux. So far so good.

When I run `make -f Makefile.mingw`, Ive got error
`make[1]: ../../win32-dev/mingw/bin/gcc: Command not found`

When I tried mingw installed (which I can use to compile pidgin itself)

CC=/usr/bin/i586-mingw32msvc-cc
```
make[1]: Entering directory `/home/spicajir/Documents/pidgin/win32-dev/telegram-purple/tgl'
/usr/bin/i586-mingw32msvc-cc -shared -o libs/libtgl.dll objs/mtproto-common.o objs/mtproto-client.o objs/mtproto-key.o objs/queries.o objs/structures.o objs/binlog.o objs/tgl.o objs/updates.o objs/tg-mime-types.o objs/mtproto-utils.o objs/crypto/bn_openssl.o objs/crypto/bn_altern.o objs/crypto/rsa_pem_openssl.o objs/crypto/rsa_pem_altern.o objs/crypto/md5_openssl.o objs/crypto/md5_altern.o objs/crypto/sha_openssl.o objs/crypto/sha_altern.o objs/crypto/aes_openssl.o objs/crypto/aes_altern.o objs/tools.o objs/crypto/rand_openssl.o objs/crypto/rand_altern.o objs/crypto/err_openssl.o objs/crypto/err_altern.o objs/auto/auto-skip.o objs/auto/auto-fetch.o objs/auto/auto-store.o objs/auto/auto-autocomplete.o objs/auto/auto-types.o objs/auto/auto-fetch-ds.o objs/auto/auto-free-ds.o objs/auto/auto-store-ds.o objs/auto/auto-print-ds.o -L../../win32-dev/libgcrypt-1.6.3/lib -I../../win32-dev/libgpg-error-1.12-2/lib -L../../win32-dev/libwebp-0.4.3-1/lib -L../../win32-dev/glib-2.28.8/lib -lwebp -lz -lgcrypt -Wl,--export-all-symbols -ggdb  -ggdb -lz -lgcrypt  
/usr/lib/gcc/i586-mingw32msvc/4.2.1-sjlj/../../../../i586-mingw32msvc/bin/ld: cannot find -lwebp
collect2: ld returned 1 exit status
```

CC=i686-w64-mingw32-gcc
```
make[1]: Entering directory `/home/spicajir/Documents/pidgin/win32-dev/telegram-purple/tgl'
i686-w64-mingw32-gcc -shared -o libs/libtgl.dll objs/mtproto-common.o objs/mtproto-client.o objs/mtproto-key.o objs/queries.o objs/structures.o objs/binlog.o objs/tgl.o objs/updates.o objs/tg-mime-types.o objs/mtproto-utils.o objs/crypto/bn_openssl.o objs/crypto/bn_altern.o objs/crypto/rsa_pem_openssl.o objs/crypto/rsa_pem_altern.o objs/crypto/md5_openssl.o objs/crypto/md5_altern.o objs/crypto/sha_openssl.o objs/crypto/sha_altern.o objs/crypto/aes_openssl.o objs/crypto/aes_altern.o objs/tools.o objs/crypto/rand_openssl.o objs/crypto/rand_altern.o objs/crypto/err_openssl.o objs/crypto/err_altern.o objs/auto/auto-skip.o objs/auto/auto-fetch.o objs/auto/auto-store.o objs/auto/auto-autocomplete.o objs/auto/auto-types.o objs/auto/auto-fetch-ds.o objs/auto/auto-free-ds.o objs/auto/auto-store-ds.o objs/auto/auto-print-ds.o -L../../win32-dev/libgcrypt-1.6.3/lib -I../../win32-dev/libgpg-error-1.12-2/lib -L../../win32-dev/libwebp-0.4.3-1/lib -L../../win32-dev/glib-2.28.8/lib -lwebp -lz -lgcrypt -Wl,--export-all-symbols -ggdb  -ggdb -lz -lgcrypt  
/usr/bin/i686-w64-mingw32-ld: cannot find -lwebp
/usr/bin/i686-w64-mingw32-ld: cannot find -lz
/usr/bin/i686-w64-mingw32-ld: cannot find -lgcrypt
/usr/bin/i686-w64-mingw32-ld: cannot find -lz
/usr/bin/i686-w64-mingw32-ld: cannot find -lgcrypt
collect2: error: ld returned 1 exit status
```

Can anyone help me here? :( I found some libraries here: http://lrn.no-ip.info/packages/i686-w64-mingw (not tested yet), but lz is mystery for me..

zlib is installed:
```
ls -la /usr/lib/i386-linux-gnu/libz.so
lrwxrwxrwx 1 root root 33 May 13  2013 /usr/lib/i386-linux-gnu/libz.so -> /lib/i386-linux-gnu/libz.so.1.2.8

apt-get install libz-dev
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Note, selecting 'zlib1g-dev' instead of 'libz-dev'
zlib1g-dev is already the newest version.
```
I even compiled zlib 1.2.11, tried both dynamic and --static, 
```
ls -la /usr/local/lib/ | grep libz
-rw-r--r--  1 root root    127606 Apr 30 22:05 libz.a
lrwxrwxrwx  1 root root        14 Apr 30 22:05 libz.so -> libz.so.1.2.11
lrwxrwxrwx  1 root root        14 Apr 30 22:05 libz.so.1 -> libz.so.1.2.11
-rwxr-xr-x  1 root root    116033 Apr 30 22:05 libz.so.1.2.11
```

Still the same error: i686-w64-mingw32-ld: cannot find -lz

**Comment by smirnov-d-a** [comment]
Do you compile it under Windows (using Cygwin) or cross-compile under Linux?
I've tried to compile it under Windows and I seemingly managed to get around missing libraries somehow.  However, my attempts to go on further were unsuccessful.

**Comment by JimmyS83** [comment]
Hi, I tried both. On Windows I tried bot cygwin / minGW.. Unfortunatelly still no luck. Eion Robb wrote me, that tl-generate dependencies we need to compile it under linux.

**Comment by BenWiederhake** [comment]
Will be included by https://github.com/majn/telegram-purple/pull/397
With that, you can do exactly what you currently need.
If you need it "just for now", it should suffice.

But see Eion's feedback, there's lots I still need to do.  For example: it currently only builds 32-bit.
The 32-bit builds (no matter whether Linux, Windows, Apple, whatever) generally are extremely slow to connect.  (The factorization proof-of-work thing is just painful on 32-bit.)

And yup, tgl is *a* problem (wants to run some self-built binaries as part of the build).  However, the problem you're currently facing (needing both i686-linux-gnu zlib and i686-w64-mingw32 zlib) has nothing to do with that.  There's lots more problems waiting, among them: i686-w64-mingw32 libwebp :)

**Comment by BenWiederhake** [comment]
Can you try out #446 for me?  There I think I listed all the dependencies in the README, and it should build rather reliably, including webp.  (Still only 32-bit, but whatever.)

**Comment by BenWiederhake** [comment]
Should be fixed in [v1.4.1](https://github.com/majn/telegram-purple/releases/tag/v1.4.1).
